File: Palindrome_Rearranging.js
// Arcade > Intro > 18
// Given a string, find out if its characters can be rearranged to form a palindrome
// Ex input = "aabb" => true
// Ex input = "abaa" => false

function palindromeRearranging(s) {
    s=s.split("") //create array of string chars
    let a = [...new Set(s)] //create set of chars
    let b = a.map(x => [x, (s.filter(y => y === x).length)%2]) //map chars to their count (0=even, 1=odd)
    let q=0
    for (let i=0; i<b.length; i++) { //iterate through map
        if (b[i][1]===1) q++ //if a char appears an odd number of times, q++
        if (q>1) return false
    }
    return true 
}

File: Palindrome_Rearranging.py
# Arcade > Intro > 18
# Given a string, find out if its characters can be rearranged to form a palindrome
# Ex input = "aabb" => true
# Ex input = "abaa" => false

def palindromeRearranging(s):
    a=0
    for i in set(s): #iterate through set of s chars
        if s.count(i)%2 != 0: #if char appears an odd number of times
            a+=1
        if a>1:
            return False
    return True